.github/workflows/test.yml

```diff
@@ -43,17 +45,12 @@ jobs:
     runs-on: ${{ matrix.platform }}-latest
     steps:
 
-    - uses: actions/setup-go@v3
-      with:
-        go-version: 1.${{ matrix.go }}.x
-
     - uses: actions/checkout@v3
```

Comment:
Is this chunk redundant now?

Reply:
I don't understand. This PR moves action/setup-go to be run after actions/checkout, and actions/cache is removed. actions/checkout is not modified. It is required in order to clone the repo.